napari/_qt/qt_event_loop.py

```diff
@@ -194,46 +201,54 @@ def gui_qt(*, startup_logo=False, gui_exceptions=False, force=False):
     IPython with the Qt GUI event loop enabled by default by using
     ``ipython --gui=qt``.
     """
-    splash_widget = None
 
     app = get_app()
+    splash = None
     if startup_logo and app.applicationName() == 'napari':
-        pm = QPixmap(NAPARI_ICON_PATH).scaled(
-            360, 360, Qt.KeepAspectRatio, Qt.SmoothTransformation
-        )
-        splash_widget = QSplashScreen(pm)
-        splash_widget.show()
-        app._splash_widget = splash_widget
+        from .widgets.qt_splash_screen import NapariSplashScreen
 
-    # instantiate the exception handler
-    exception_handler = ExceptionHandler(gui_exceptions=gui_exceptions)
-    sys.excepthook = exception_handler.handle
+        splash = NapariSplashScreen()
+        splash.close()
+
+    yield app
+    run(force=force, gui_exceptions=gui_exceptions, _func_name='gui_qt')
 
+
+def _ipython_has_eventloop() -> bool:
+    """Return True if IPython %gui qt is active.
+
+    Using this is better than checking ``QApp.thread().loopLevel() > 0``,
+    because IPython starts and stops the event loop continuously to accept code
+    at the prompt.  So it will likely "appear" like there is no event loop
+    running, but will still don't need to start one.
+    """
     try:
-        yield app
-    except Exception:
-        exception_handler.handle(*sys.exc_info())
-
-    # if the application already existed before this function was called,
-    # there's no need to start it again.  By avoiding unnecessary calls to
-    # ``app.exec_``, we avoid blocking.
-    # we add 'magicgui' so that anyone using @magicgui *before* calling gui_qt
-    # will also have the application executed. (a bandaid for now?...)
-    # see https://github.com/napari/napari/pull/2016
-    if app.applicationName() in ('napari', 'magicgui'):
-        if splash_widget and startup_logo:
-            splash_widget.close()
-        run(force=force, _func_name='gui_qt')
-
-
-def run(*, force=False, _func_name='run'):
+        from IPython import get_ipython
```

Comment:
Nice that this is inside a try catch as I think we're about to drop the `IPython` dependency in #2118. I was worried for a second before seeing that!!